app/models/user.rb

```diff
@@ -4,21 +4,24 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable, :confirmable,
          :recoverable, :rememberable, :trackable, :validatable
 
-  has_many :pending_purchases
+  has_many :line_items
+  has_one :cart
+
+  after_save :create_cart
+
+  def create_cart
+    Cart.create!(user: self)
+  end
 
   def add_to_cart(book)
```

Comment:
Thinking about OO design, this method and the two below it look to me like they should be calling methods from the cart class not implementing the logic themselves. Something more like:
```
def add_to_cart(book)
  cart.add(book)
end
def remove_from_cart(book)
  cart.remove(book)
end

def in_cart?(book)
  cart.contains?(book)
end
```
Then you would move the logic to the cart class. This would reduce the amount that the user class needs to know about the cart class (now the user doesn't need to know anything about line items) and allow the cart to be responsible for how things behave inside of it. This idea is pretty closely related to some of the ideas in the first few chapters of POODR if you're looking for a reference.